### PR TITLE
Serialization for Currency

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

Add support for serializing Currency objects.

### Context of Change

We currently don't serialize all fields in transactions. This is a step towards fixing that. 

Currencies are serialized as either a hex encoded string or a currency string. See: https://xrpl.org/currency-formats.html#currency-codes

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

No change since this isn't hooked up to anything yet. Future PRs will build on this to add in support for Paths in payments. 

## Test Plan

CI; new tests provided.

### CODE REVIEW

File: src/XRP/serializer.ts [hunk 1]
```diff
@@ -77,6 +77,7 @@ interface PathElementJSON {
   currencyCode?: string
 }
 
+type CurrencyJSON = string
```
Comment (amiecorso) [inline src/XRP/serializer.ts:80]: does a naked string count as JSON?  news to me, but seems reasonable

**Review by amiecorso — APPROVED** [review]

File: src/XRP/serializer.ts (hunk 1 above)
Comment (Stormtv) [reply src/XRP/serializer.ts:80]: Yeah, I think this type is unnecessary since the code referencing this is returning hex in a string format.

**Review by Stormtv — APPROVED** [review]
Looks good but I would recommend removing the CurrencyJSON type and using string

**Comment by keefertaylor** [comment]
I'd prefer to have this stuff typed for safety / readability.

**Pull request merged**